src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -404,32 +427,37 @@ function genFragment(
     lastList = nodeName;
   }
 
+  const blockType = getBlockTypeForTag(nodeName, lastList, blockRenderMap);
+
   // Block Tags
-  if (!inBlock && blockTags.indexOf(nodeName) !== -1) {
-    chunk = getBlockDividerChunk(
-      getBlockTypeForTag(nodeName, lastList, blockRenderMap),
-      depth,
-    );
+  if (lastList && inBlock === 'li' && nodeName === 'li') {
+    chunk = getBlockDividerChunk(blockType, depth, parentKey);
+    blockKey = chunk.blocks[0].key;
     inBlock = nodeName;
-    newBlock = true;
-  } else if (lastList && inBlock === 'li' && nodeName === 'li') {
+    nextBlockType =
+      lastList === 'ul' ? 'unordered-list-item' : 'ordered-list-item';
+    newBlock = !experimentalTreeDataSupport;
+  } else if (
+    (!inBlock || experimentalTreeDataSupport) &&
+    blockTags.indexOf(nodeName) !== -1
+  ) {
     chunk = getBlockDividerChunk(
       getBlockTypeForTag(nodeName, lastList, blockRenderMap),
       depth,
+      parentKey,
     );
+    blockKey = chunk.blocks[0].key;
     inBlock = nodeName;
-    newBlock = true;
-    nextBlockType =
-      lastList === 'ul' ? 'unordered-list-item' : 'ordered-list-item';
+    newBlock = !experimentalTreeDataSupport;
```

Comment:
This is the same as the above, except that the above one create list blocks and this create all the other blocks